### PR TITLE
ModuleTree and SLangCheck interfaces

### PR DESCRIPTION
**Pull request by marcelo-sousa-sonarsource**

### CODE REVIEW

File: slang-api/src/main/java/com/sonarsource/slang/api/ModuleTree.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.sonarsource.slang.api;
+
+import java.util.List;
+
+/**
+ * SLang module unit.
+ */
+public interface ModuleTree extends Tree {
```
Comment (pynicolas) [inline slang-api/src/main/java/com/sonarsource/slang/api/ModuleTree.java]: I don't think we need that for the moment.

File: slang-api/src/main/java/com/sonarsource/slang/api/ModuleTree.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  List<Tree> types();
```
Comment (pynicolas) [inline slang-api/src/main/java/com/sonarsource/slang/api/ModuleTree.java]: That seems to indicate that we don't support statements at the top level.

File: slang-api/src/main/java/com/sonarsource/slang/api/SLangFileScannerContext.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package com.sonarsource.slang.api;
+
+import java.io.File;
+import java.util.Objects;
+
+/**
+ * Context injected in check classes and used to report issues.
+ */
+public interface SLangFileScannerContext {
```
Comment (pynicolas) [inline slang-api/src/main/java/com/sonarsource/slang/api/SLangFileScannerContext.java]: That should be part of #3.

**Comment by marcelo-sousa-sonarsource** [comment]
These changes have already been incorporated in other pull requests.

**Pull request closed without merging**